alarms/alarms.py

```diff
@@ -271,17 +303,20 @@ def _raise_alarm(self, alarm, check_string, arguments):
         last_alarm_time = self._get_time_of_last_alarm(alarm['id'])
 
         # Alarm is not inhibited by no_repeat_interval
-        if last_alarm_time is None or time.time() - last_alarm_time > alarm['no_repeat_interval']:
+        if (
+                last_alarm_time is None or
+                time.time() - last_alarm_time > alarm['no_repeat_interval']
+        ):
             if last_alarm_time is None:
                 _LOG.debug('No previous alarm within no_repeat_interval. '
                            'No previous alarm.')
             else:
                 diff = time.time() - last_alarm_time
                 _LOG.debug('No previous alarm within no_repeat_interval. Time '
                            'since last: {0:.1f}'.format(diff))
-
-            query = "INSERT INTO alarm_log (alarm_id) VALUES (%s)"
-            self._alarm_cursor.execute(query, (alarm['id']))
+            # Todo: Consider to write a prepared statement rather than using format
+            query = 'INSERT INTO alarm_log set alarm_id={}'.format(alarm['id'])
```

Comment:
I'm a little in doubt here. The queries seems to have changes. Possibly something about whether there could in principle have been more than one alarms triggered and logged maybe. It seems to have been a bug in original code however, since there will clearly only be inserted on item.

Reply:
Adding a comma to declare the second argument a tuple fixed the syntax for me:
`self._alarm_cursor.execute(query, (alarm['id'],))`
Was that a Python2 -> Python3 problem?

Reply:
Ok. So my comment was a little unclear. The point of my comment was that originally I thought there was a bug here, because it seemed that you had changed the query code from handling multiple to just one row, but then I reread the code and realized that it must have been a bug in my original code to ever make that query for multiple rows, since the code clearly only handles a single id. So, long story short, my bad and thanks for fixing.